.pipelines/e2e.yml

```diff
@@ -10,47 +10,124 @@ resources:
-  - script: |
-      az account set -s $AZURE_SUBSCRIPTION_ID
-      export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
-      make secrets
-      . secrets/env
-      echo "##vso[task.setvariable variable=RP_MODE]$RP_MODE"
-    displayName: 🔑 Downloading certificates and secrets from storage account
-    name: setEnv
-  - template: ./templates/template-push-images-to-acr.yml
-    parameters:
-      rpImageACR: $(RP_IMAGE_ACR)
-      buildCommand: publish-image-aro
-  - script: |
-      set -e
-      set -o pipefail
-
-      . secrets/env
-
-      az account set -s $AZURE_SUBSCRIPTION_ID
-
-      set -x
-      . ./hack/e2e/run-rp-and-e2e.sh
-      trap 'set +e; kill_rp; clean_e2e_db' EXIT
-
-      deploy_e2e_db
-
-      run_rp
-      validate_rp_running
-      register_sub
-
-      export CI=true
-      make test-e2e
-  - template: ./templates/template-az-cli-logout.yml
+
+# Run the test suite
+stages:
+- stage: E2ESuite
+  jobs:
+  - job: Execute_Tests
+    timeoutInMinutes: 180
+    pool:
+      name: ARO-CI
+      demands: go-1.16
+    steps:
+    - template: ./templates/template-checkout.yml
+    - template: ./templates/template-az-cli-login.yml
+      parameters:
+        azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
+    - template: ./templates/template-certificates-and-secrets.yml
+    - template: ./templates/template-push-images-to-acr.yml
+      parameters:
+        rpImageACR: $(RP_IMAGE_ACR)
+        buildCommand: publish-image-aro
+    - script: |
+        set -e
+        set -o pipefail
+
+        . secrets/env
+        az account set -s $AZURE_SUBSCRIPTION_ID
+
+        set -x
+        . ./hack/e2e/run-rp-and-e2e.sh
+        trap 'set +e; kill_rp' EXIT
+
+        deploy_e2e_db
+        run_rp
+        validate_rp_running
+        register_sub
+
+        export CI=true
+        make test-e2e
+      displayName: Run E2E suite
+
+# Collect and publish must-gather if test suite fails
+- stage: MustGather
+  dependsOn: E2ESuite
+  condition: failed()
+  jobs:
+  - job: Collect_Must_Gather
+    timeoutInMinutes: 30
+    pool:
+      name: ARO-CI
+      demands: go-1.16
+    steps:
+    - template: ./templates/template-az-cli-login.yml
+      parameters:
+        azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
+    - template: ./templates/template-certificates-and-secrets.yml
+    - template: ./templates/template-push-images-to-acr.yml
+      parameters:
+        rpImageACR: $(RP_IMAGE_ACR)
+        buildCommand: publish-image-aro
+    - script: |
+        set -e
+        set -o pipefail
+
+        . secrets/env
+        az account set -s $AZURE_SUBSCRIPTION_ID
+
+        set -x
+        . ./hack/e2e/run-rp-and-e2e.sh
+        trap 'set +e; kill_rp' EXIT
+
+        run_rp
+        validate_rp_running
+        register_sub
+
+        wget https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.9.8/openshift-client-linux-4.9.8.tar.gz
+        tar xf openshift-client-linux-4.9.8.tar.gz
+
+        make admin.kubeconfig
```

Comment:
instead of making a new admin.kubeconfig, can we try to store the admin.kubeconfig from E2E suite stage and reference that here?